### DISCUSSION TITLE
Playlist txt song entries

### DISCUSSION THREAD

**Discussion by Brandon23031** [jagrosh/MusicBot#890] [Questions]
Hi 

I tried creating a playlist from a list of artist and song names, but it seems only url's to songs work.
Is this the only way that the playlist txt can be populated? By using url's?

If so, does anyone have any tips on quickly turning a big list of songs into a list of urls?

Thanks in advance!

**Comment by MichailiK** [comment]
Relative paths should also work.

**Reply by Brandon23031** [reply]
Please do explain...

**Reply by MichailiK** [reply]
You can play any audio file you put within your JMusicBot.

Let's say you have a folder called "Songs" in your JMusicBot folder:
```
|- JMusicBot-0.3.5.jar
|- config.txt
|- Songs
|  |- (empty folder)
```
Now let's add some music in that folder, like `Lifelight.mp3`, `By Your Side.mp3` and `Aint Nobody.mp3` :
```
|- JMusicBot-0.3.5.jar
|- config.txt
|- Songs
|  |- Lifelight.mp3
|  |- By Your Side.mp3
|  |- Aint Nobody.mp3
```
You can then make a playlist.txt-file so:
```
Songs/Lifelight.mp3
Songs/By Your Side.mp3
Songs/Aint Nobody.mp3
```
(You need to use a slash after folder names)

JMusicBot should go inside the "Songs" folder and load all 3 songs you've specified.

**Reply by Brandon23031** [reply]
Ooooh so this is a workaround for if you have the songs on your machine (if I understand correctly). Sadly I only have the songs as a list on excel, any ideas for that scenario?

**Reply by MichailiK** [reply]
another thing you can do is prepend `ytsearch:` to all of your items. like `ytsearch:Lifelight` or `ytsearch:By Your Side` to find them on YouTube.

**Reply by Brandon23031** [reply]
Awesome will try that, thanks!

**Reply by Brandon23031** [reply]
Awesome, it plays the songs! Does it still struggle with playlists longer than 128 entries?

**Reply by MichailiK** [reply]
It shouldn't. It may take a short while to fill the queue though, depending on how big it is. That's because the bot fetches all YouTube videos one by one in the background.

**Reply by Brandon23031** [reply]
Thanks appreciate it!
